static/app/views/dashboardsV2/index.tsx

```diff
@@ -1,27 +1,61 @@
 import * as React from 'react';
 import {RouteComponentProps} from 'react-router';
 
+import {Client} from 'app/api';
 import Feature from 'app/components/acl/feature';
+import NotFound from 'app/components/errors/notFound';
+import LoadingIndicator from 'app/components/loadingIndicator';
 import {Organization} from 'app/types';
+import withApi from 'app/utils/withApi';
 import withOrganization from 'app/utils/withOrganization';
 
-import Detail from './detail';
+import DashboardDetail from './detail';
+import OrgDashboards from './orgDashboards';
+import {DashboardState} from './types';
 
-type Props = RouteComponentProps<{orgId: string; dashboardId: string}, {}> & {
+type Props = RouteComponentProps<{orgId: string}, {}> & {
+  api: Client;
   organization: Organization;
   children: React.ReactNode;
 };
 
 class DashboardsV2Container extends React.Component<Props> {
   render() {
-    const {organization, ...props} = this.props;
+    const {organization, params, api, location, children} = this.props;
+
+    if (!organization.features.includes('dashboards-manage')) {
+      return (
+        <OrgDashboards
+          api={api}
+          location={location}
+          params={params}
+          organization={organization}
+        >
+          {({dashboard, dashboards, error, reloadData}) => {
+            return error ? (
+              <NotFound />
+            ) : dashboard ? (
+              <DashboardDetail
+                {...this.props}
+                initialState={DashboardState.VIEW}
+                dashboard={dashboard}
+                dashboards={dashboards}
+                reloadData={reloadData}
+              />
+            ) : (
+              <LoadingIndicator />
+            );
+          }}
+        </OrgDashboards>
+      );
+    }
 
     return (
-      <Feature features={['dashboards-basic']} organization={organization}>
-        <Detail {...props} organization={organization} />
+      <Feature features={['dashboards-manage']} organization={organization}>
```

Comment:
should we remove this feature component, since we are checking this information above?